src/peft/tuners/xlora/model.py

```diff
@@ -0,0 +1,345 @@
+            model_peft.generate,
+        )
+        model_peft.save_pretrained = peft_model_wrapper.save_pretrained  # type: ignore
+        model_peft.generate = peft_model_wrapper.generate  # type: ignore
+
+        # Setup the model internal state
+        self.internal_xlora_classifier = xlora_classifier
+        self.internal_xlora_scalings = None  # type: ignore
+        self.xlora_config = peft_config
+
+    def set_topk_lora(self, value: Optional[int]):
+        """
+        Sparsely select the specified top_k LoRA experts instead of the default dense method. Set to None to use dense. This is reflected in the config.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.config.top_k_lora = value
+
+    def get_topk_lora(self) -> Optional[int]:
+        """
+        Get the current top_k LoRA experts value.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        return classifier.config.top_k_lora
+
+    def set_global_scaling_weight(self, weight: float):
+        """
+        Set the global LoRA weight, a scalar to multiply the output of each LoRA adapter by. This is by default 1. This is reflected in the config.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.config.global_scaling_weight = weight
+
+    def get_global_scaling_weight(self) -> float:
+        """
+        Get the global LoRA weight.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        return classifier.config.global_scaling_weight
+
+    def get_latest_scalings(self) -> Optional[torch.Tensor]:
+        """
+        Returns the latest scalings prediction, or None if no scalings have been predicted. The tensor is of shape (batch_size, seq_len, n_layers, n_classes).
+        """
+        return self.internal_xlora_scalings
+
+    def get_scalings_log(self) -> List[torch.Tensor]:
+        """
+        Returns a shallow (only copying the list itself not the tensors) copy of the list containing the scalings log. Editing the list does not change the underlying log.
+        The tensors are of shape (batch_size, seq_len, n_layers, n_classes). The seq_len dim may vary with input dimension.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        return classifier.log_scalings.copy()
+
+    def set_scaling_pass_value(self, value: Union[Number, None]):
+        """
+        Manually set the scalings to a specific value during the scaling pass, forever. Call this function with None to enable the default
+        scalings.
+
+        This is reflected in the config.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.set_override_scaling_pass_value(value)
+
+    def print_scalings_predictions(self, n_predictions_lifetime: int):
```

Comment:
Hmm, what is printed here?

Reply:
It is connected to the classifier, specifically [this](https://github.com/huggingface/peft/pull/1491/files#diff-69a113f38e9fa6b36a24cd8880a9847d80ea7a0f2e9bc89acbed9084504e66ffR182-R184) part. However, this is more of a debugging API (it is also redundant to the scalings log) and I think it would be OK to remove.

Reply:
Yes, let's remove the debugging functionality here. I don't think we have anything like this elsewhere in PEFT. This should also allow to remove `n_predictions_lifetime` entirely, right?

Reply:
Yes, this does. I have removed it.